Minor protectedFoo() cleanup

annevk opened the pull request:
#### 23ff6ac20106fafaff3b0411e460890239dec0c1
<pre>
Minor protectedFoo() cleanup
<a href="https://bugs.webkit.org/show_bug.cgi?id=294158">https://bugs.webkit.org/show_bug.cgi?id=294158</a>
<a href="https://rdar.apple.com/152771093">rdar://152771093</a>

Reviewed by Chris Dumez.

Remove a couple of protectedFoo() methods I wasn&apos;t sure about before.
The underlying m_foo is const so it seems better to use an explicit
Ref/RefPtr constructor where needed then to rely on protectedFoo() to
reduce the chance of people accidentally using those where they are not
needed.

Canonical link: <a href="https://commits.webkit.org/295962@main">https://commits.webkit.org/295962@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/cfc70da8165fee2e06b0330222b397343f4afc31

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/106728 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/26479 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/16876 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/111939 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/57326 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/108767 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/27149 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/34981 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/81034 "Passed tests") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/109732 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/21542 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/96307 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/61374 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/20980 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/14414 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/56780 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/90882 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/14443 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/114857 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/33866 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/24963 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/90101 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/34231 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/92538 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/89811 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/34745 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/12553 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/29509 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/17287 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/33791 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/39204 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/33537 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/36890 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/35136 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

cdumez commented on Source/WebCore/page/Navigation.cpp:
```diff
@@ -912,7 +912,7 @@ Navigation::DispatchResult Navigation::innerDispatchNavigateEvent(NavigationNavi
         navigationType,
         destination.ptr(),
-        abortController->protectedSignal(),
+        Ref { abortController->signal() },
```
Do we really need the `Ref { }` here?

annevk replied:
Yes, for this one and the two you flagged below I could not get it to compile without the explicit wrapper unfortunately. That's why I kept around the protected methods initially.

cdumez commented on Source/WebCore/animation/ViewTimeline.cpp:
```diff
@@ -102,10 +102,10 @@ ExceptionOr<ViewTimeline::SpecifiedViewTimelineInsets> ViewTimeline::validateSpe
             return { {
-                dynamicDowncast<CSSPrimitiveValue>(insetPair->protectedFirst()),
-                dynamicDowncast<CSSPrimitiveValue>(insetPair->protectedSecond())
+                RefPtr { dynamicDowncast<CSSPrimitiveValue>(insetPair->first()) },
```
Do we really need the `RefPtr { }` here?

cdumez commented on Source/WebCore/animation/ViewTimeline.cpp:
```diff
@@ -102,10 +102,10 @@ ExceptionOr<ViewTimeline::SpecifiedViewTimelineInsets> ViewTimeline::validateSpe
-                dynamicDowncast<CSSPrimitiveValue>(insetPair->protectedFirst()),
-                dynamicDowncast<CSSPrimitiveValue>(insetPair->protectedSecond())
+                RefPtr { dynamicDowncast<CSSPrimitiveValue>(insetPair->first()) },
+                RefPtr { dynamicDowncast<CSSPrimitiveValue>(insetPair->second()) }
```
Ditto.

annevk committed:
Minor protectedFoo() cleanup
https://bugs.webkit.org/show_bug.cgi?id=294158
rdar://152771093

Reviewed by Chris Dumez.

Remove a couple of protectedFoo() methods I wasn't sure about before.
The underlying m_foo is const so it seems better to use an explicit
Ref/RefPtr constructor where needed then to rely on protectedFoo() to
reduce the chance of people accidentally using those where they are not
needed.

Canonical link: https://commits.webkit.org/295962@main
